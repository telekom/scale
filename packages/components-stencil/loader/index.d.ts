
export * from '../dist/types/components';
export interface CustomElementsDefineOptions {
  exclude?: string[];
  resourcesUrl?: string;
  syncQueue?: boolean;
  raf?: (c: FrameRequestCallback) => number;
  ael?: (el: EventTarget, eventName: string, listener: EventListenerOrEventListenerObject, options: boolean | AddEventListenerOptions) => void;
  rel?: (el: EventTarget, eventName: string, listener: EventListenerOrEventListenerObject, options: boolean | AddEventListenerOptions) => void;
}
export declare function defineCustomElements(win: Window, opts?: CustomElementsDefineOptions): Promise<void>;
export declare function applyPolyfills(): Promise<void>;
