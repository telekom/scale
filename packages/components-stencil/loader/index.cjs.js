
  module.exports = require('../dist/cjs/loader.cjs.js');
  module.exports.applyPolyfills = function() { return Promise.resolve() };
  