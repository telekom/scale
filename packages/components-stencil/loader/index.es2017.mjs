export * from '../dist/esm/polyfills/index.js';
export * from '../dist/esm/loader.mjs';