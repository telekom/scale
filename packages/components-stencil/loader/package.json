{
  "name": "telements-components-stencil-loader",
  "typings": "./index.d.ts",
  "module": "./index.mjs",
  "main": "./index.cjs.js",
  "jsnext:main": "./index.es2017.mjs",
  "es2015": "./index.es2017.mjs",
  "es2017": "./index.es2017.mjs"
}