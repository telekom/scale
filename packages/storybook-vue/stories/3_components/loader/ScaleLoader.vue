<template>
  <scale-loader
    :variant="variant"
    :alignment="alignment"
    :text="text"
    :size="size"
  >
  </scale-loader>
</template>

<script>
import { action } from '@storybook/addon-actions';

export default {
  props: {
    variant: { type: 'white' | 'primary', default: 'primary' },
    alignment: {
      type: 'horizontal' | 'vertical',
      default: 'horizontal',
    },
    text: { type: String },
    size: { type: 'small' | 'large', default: 'small' },
  },
};
</script>
