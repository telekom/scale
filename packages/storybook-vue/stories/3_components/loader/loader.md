<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Loader</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

This component contains the Telekom loader in several variants: Different colors, sizes, without text and with a variable text.

## Beta components

This component is still in the beta phase. When testing it, keep in mind that it may not have gone through all quality control measures, and it may not yet have WCAG accessibility certification. There may be changes to this component in the future.
