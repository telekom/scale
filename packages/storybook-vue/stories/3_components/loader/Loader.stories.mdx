import { Meta, Story, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import ScaleLoader from './ScaleLoader.vue';
import { action } from '@storybook/addon-actions';

<Meta title="Beta Components/Loader" component={ScaleLoader} argTypes={{}} />

export const Template = (args, { argTypes }) => ({
  components: { ScaleLoader },
  props: {
    label: String,
    ...ScaleLoader.props,
  },
  template: `
    <scale-loader
       :variant="variant"
    :alignment="alignment"
    :text="text"
    :size="size"
    >
    </scale-loader>
    `,
});

# Loader

## Standard

<Canvas withSource="none">
  <Story name="Standard" args={{}}>
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />

## Large Size & Text

<Canvas withSource="none">
  <Story
    name="White"
    args={{
      size: 'large',
      text: 'Loading...',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-logo size="large" text="Loading..."></scale-logo>
```
