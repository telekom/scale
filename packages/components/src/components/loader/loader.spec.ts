import { newSpecPage } from '@stencil/core/testing';
import { Loader } from './loader';

describe('Loader', () => {
  let element;
  beforeEach(async () => {
    element = new Loader();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Loader],
      html: `<scale-loader></scale-loader>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.alignment = 'vertical';
    expect(element.getCssClassMap()).toContain('spinner--alignment-vertical');
    element.variant = 'white';
    expect(element.getCssClassMap()).toContain('spinner--variant-white');
    element.size = 'large';
    expect(element.getCssClassMap()).toContain('spinner--size-large');
    element.text = 'Loading';
    expect(element.getCssClassMap()).toContain('spinner--text');
  });
});
