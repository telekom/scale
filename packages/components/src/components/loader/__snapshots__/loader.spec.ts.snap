// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<scale-loader>
  <mock:shadow-root>
    <div class="spinner spinner--alignment-horizontal spinner--size-small spinner--variant-magenta" part="spinner alignment-horizontal variant-magenta size-small">
      <div class="spinner__container" part="container">
        <div class="spinner__circle-background"></div>
        <div class="spinner__circle" part="circle">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      </div>
      <div aria-busy="true" aria-live="polite" class="spinner__text" part="text"></div>
    </div>
  </mock:shadow-root>
</scale-loader>
`;
